### PR TITLE
SCP-2677 builtin data for plutus-metatheory + upstream builtin updates

### PR DESCRIPTION
**Pull request by jmchapman**
I would like to add at least minimal support for this so I can process untyped validators that make use of it for testing.

metatheory currently doesn't support polymorphic type constants for lists and pairs. It does support polymorphic builtin functions.

It appears that I need to add polymorphic type constants as for data I need at least (1) lists of data and (2) lists of pairs of data and I don't want to duplicate the list machinery

Pre-submit checklist:
- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by jmchapman** [comment]
there have been some upstream builtin changes which I will need to merge into this.

**Comment by jmchapman** [comment]
ok, it typechecks in Agda

**Comment by jmchapman** [comment]
merged

**Comment by jmchapman** [comment]
It appears to me that the plutus-metatheory tests are passing. Not all builtins are supported yet though.

But this PR is failing in CI on seemingly unrelated things.

**Comment by michaelpj** [comment]
Looks like your `updateMaterialized` didn't quite do the job somehow?

File: cabal.project [hunk 1]
```diff
@@ -74,7 +74,7 @@ package cardano-api
 
 -- Turn off the tests for a while
 package plutus-metatheory
-  tests: False
+  tests: True
```
Comment (michaelpj) [inline cabal.project:77]: Just delete this stanza entirely

**Comment by jmchapman** [comment]
thanks @michaelpj, that seems to have done it. That gives me a solid base to build on!

**Pull request merged**